pkg/authn/challenge/challenge_ns_anon.rego

```diff
@@ -0,0 +1,40 @@
+package bacalhau.authn
+
+import rego.v1
+
+# Implements a policy where any clients with a client ID are permitted access.
+# Anonymous users will authorised as long as they can generate a key pair.
+#
+# Modify the `ns` key of the token to control what namespaces they can access.
+
+now := time.now_ns() / 1000
+
+one_month := time.add_date(time.now_ns(), 0, 1, 0) / 1000
+
+token := io.jwt.encode_sign(
+	{
+		"typ": "JWT",
+		"alg": "RS256",
+	},
+	{
```

Comment:
Are we missing a subject?  Even the fingerprint might do if we're not tying it to a human (which we might via a idToken in other challenge types?).